lib/decode/decode.go

```diff
@@ -44,21 +44,28 @@ func HookTranslateKeys(_, to reflect.Type, data interface{}) (interface{}, error
 	}
 
 	rules := translationsForType(to)
+	// Avoid making a copy if there are no translation rules
+	if len(rules) == 0 {
+		return data, nil
+	}
+	result := make(map[string]interface{}, len(source))
 	for k, v := range source {
 		lowerK := strings.ToLower(k)
 		canonKey, ok := rules[lowerK]
 		if !ok {
+			result[lowerK] = v
 			continue
 		}
-		delete(source, k)
 
 		// if there is a value for the canonical key then keep it
-		if _, ok := source[canonKey]; ok {
+		if v, ok := source[canonKey]; ok {
```

Comment:
nitpick: it could be nice to rename `v` here to something like `existing` to further distinguish from the `v` from the `source`.